core/dbt/flags.py

```diff
@@ -188,20 +193,36 @@ def get_flag_value(flag, args, user_config):
     return flag_value
 
 
+def _check_mutually_exclusive(group, args, user_config):
+    set_flag = None
+    for flag in group:
+        flag_set_by_user = not _flag_value_from_default(flag, args, user_config)
+        if flag_set_by_user and set_flag:
+            raise ValueError(f"{flag.lower()}: not allowed with argument {set_flag.lower()}")
```

Comment:
This could be nicer, in cases where I've set `warn_error` in `UserConfig` and I'm also passing the conflicting `--warn-error-options` flag on the CLI:
```
ValueError: warn_error_options: not allowed with argument warn_error
```
Sounds like it will be easier to handle in the new `click` CLI. It serves its purpose for now, and I'm happy with making it nicer in v1.5+